multiqc/modules/cutadapt/cutadapt.py

```diff
@@ -142,31 +146,44 @@ def parse_cutadapt_logs(self, f):
                 if "===" in l:
                     log_section = l.strip().strip("=").strip()
 
+                if "Overview of removed sequences" in l:
+                    if "' end" in l:
+                        res = re.search("(\d)' end", l)
+                        end = int(res.group(1))
+                    else:
+                        end = 0
+
+                    # Initilise dictionaries for length data if not already done
+                    if end not in self.cutadapt_length_counts:
+                        self.cutadapt_length_counts[end] = dict()
+                        self.cutadapt_length_exp[end] = dict()
+                        self.cutadapt_length_obsexp[end] = dict()
+
                 # Histogram showing lengths trimmed
                 if "length" in l and "count" in l and "expect" in l:
                     plot_sname = s_name
                     if log_section is not None:
                         plot_sname = "{} - {}".format(s_name, log_section)
-                    self.cutadapt_length_counts[plot_sname] = dict()
-                    self.cutadapt_length_exp[plot_sname] = dict()
-                    self.cutadapt_length_obsexp[plot_sname] = dict()
+                    self.cutadapt_length_counts[end][plot_sname] = dict()
+                    self.cutadapt_length_exp[end][plot_sname] = dict()
+                    self.cutadapt_length_obsexp[end][plot_sname] = dict()
+
                     # Nested loop to read this section while the regex matches
                     for l in fh:
                         r_seqs = re.search("^(\d+)\s+(\d+)\s+([\d\.]+)", l)
                         if r_seqs:
                             a_len = int(r_seqs.group(1))
-                            self.cutadapt_length_counts[plot_sname][a_len] = int(r_seqs.group(2))
-                            self.cutadapt_length_exp[plot_sname][a_len] = float(r_seqs.group(3))
+                            self.cutadapt_length_counts[end][plot_sname][a_len] = int(r_seqs.group(2))
+                            self.cutadapt_length_exp[end][plot_sname][a_len] = float(r_seqs.group(3))
                             if float(r_seqs.group(3)) > 0:
-                                self.cutadapt_length_obsexp[plot_sname][a_len] = float(r_seqs.group(2)) / float(
+                                self.cutadapt_length_obsexp[end][plot_sname][a_len] = float(r_seqs.group(2)) / float(
                                     r_seqs.group(3)
                                 )
                             else:
                                 # Cheating, I know. Infinity is difficult to plot.
-                                self.cutadapt_length_obsexp[plot_sname][a_len] = float(r_seqs.group(2))
+                                self.cutadapt_length_obsexp[end][plot_sname][a_len] = float(r_seqs.group(2))
```

Comment:
I was puzzled by the axis as well at first, but it is because when the expected value is zero Obs/Exp is equal to Counts